sdk/instance/helpers.go

```diff
@@ -77,9 +76,9 @@ func extractVolumesFrom(s *service.Service, d *service.Dependency) ([]container.
 // volumeKey creates a key for service's volume based on the sid to make sure that the volume
 // will stay the same for different versions of the service.
 func volumeKey(s *service.Service, dependency string, volume string) string {
-	return hex.EncodeToString(xstructhash.Hash([]string{
-		s.Sid,
-		dependency,
-		volume,
-	}, 1))
+	d := sha1.New()
+	d.Write([]byte(s.Sid))
+	d.Write([]byte(dependency))
+	d.Write([]byte(volume))
+	return hex.EncodeToString(d.Sum(nil))
```

Comment:
Did you try to use Hash.Hash?
A sha256 base58 encoded should be small enough for docker?

Reply:
No, I have not tested, but I can try. I thought it would be too much.

Reply:
```
        "Mounts": [
            {
                "Type": "volume",
                "Name": "8RBsoeyoRwajj86MZfZE6gMDJQVYGYcdSfx1zxqxNHbr",
                "Source": "/var/lib/docker/volumes/8RBsoeyoRwajj86MZfZE6gMDJQVYGYcdSfx1zxqxNHbr/_data",
```
It's working